File: src/types/claim.ts
export enum ClaimStatus {
  Draft = "Draft",
  Submitted = "Submitted",
  UnderReview = "UnderReview",
  Approved = "Approved",
  Rejected = "Rejected",
}

export enum ClaimType {
  Colision = "Colision",
  Robo = "Robo",
  Incendio = "Incendio",
  Cristales = "Cristales",
}

export interface CreateClaimRequest {
  policyNumber: string;
  claimDate: string;
  claimType: ClaimType;
  vehiclePlate: string;
  insuredName: string;
  phone: string;
  address: string;
  postalCode: string;
  description: string;
}

export interface ClaimResponse {
  id: string;
  policyNumber: string;
  claimDate: string;
  claimType: ClaimType;
  vehiclePlate: string;
  insuredName: string;
  phone: string;
  address: string;
  postalCode: string;
  description: string;
  status: ClaimStatus;
  createdAt: string;
}

export interface ValidationProblemDetails {
  title: string;
  status: number;
  detail: string;
  errors: Record<string, string[]>;
}

export interface FormErrors {
  policyNumber?: string;
  claimDate?: string;
  claimType?: string;
  vehiclePlate?: string;
  insuredName?: string;
  phone?: string;
  address?: string;
  postalCode?: string;
  description?: string;
}

File: backend/src/ClaimsManagement.Domain/Entities/Claim.cs
using ClaimsManagement.Domain.Enums;

namespace ClaimsManagement.Domain.Entities;

public class Claim
{
    public Guid Id { get; private set; }
    public string PolicyNumber { get; private set; } = string.Empty;
    public DateTime ClaimDate { get; private set; }
    public ClaimType ClaimType { get; private set; }
    public string VehiclePlate { get; private set; } = string.Empty;
    public string InsuredName { get; private set; } = string.Empty;
    public string Phone { get; private set; } = string.Empty;
    public string Address { get; private set; } = string.Empty;
    public string PostalCode { get; private set; } = string.Empty;
    public string Description { get; private set; } = string.Empty;
    public ClaimStatus Status { get; private set; }
    public DateTime CreatedAt { get; private set; }

    private Claim() { }

    public static Claim Create(
        string policyNumber,
        DateTime claimDate,
        ClaimType claimType,
        string vehiclePlate,
        string insuredName,
        string phone,
        string address,
        string postalCode,
        string description)
    {
        return new Claim
        {
            Id = Guid.NewGuid(),
            PolicyNumber = policyNumber,
            ClaimDate = claimDate,
            ClaimType = claimType,
            VehiclePlate = vehiclePlate,
            InsuredName = insuredName,
            Phone = phone,
            Address = address,
            PostalCode = postalCode,
            Description = description,
            Status = ClaimStatus.Draft,
            CreatedAt = DateTime.UtcNow
        };
    }
}
